protected/humhub/config/env.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+
+use yii\base\InvalidArgumentException;
+use yii\helpers\ArrayHelper;
+use yii\helpers\Inflector;
+use yii\helpers\Json;
+use yii\helpers\StringHelper;
+
+$env = [];
+
+$preparePath = function (string $key): array {
+    return ArrayHelper::getColumn(
+        explode('__', $key),
```

Comment:
Somehow the syntax e.g. `HUMHUB_SETTINGS_BASE__MAILER_...`  looks like a typo for me.

Any idea for other variants?

- Maybe `HUMHUB_SETTINGS_BASE_MAILER-DSN`   (See `_` as new array node, and replace `.`   with `-`, since a `.` is not supported?

Reply:
@luke- replaced `__` with `.`, `-` is not supported. now it's:
`HUMHUB_PARAMS.MODULE_AUTOLOAD_PATHS`=["/app/modules/humhub","/app/modules/humhub-contrib"]
`HUMHUB_SETTINGS.BASE.MAILER.DSN`="smtp://...."
`HUMHUB_SETTINGS.BASE.MAILER.TRANSPORT_TYPE`="php"
`HUMHUB_SETTINGS.BASE.MAILER.SYSTEM_EMAIL_ADDRESS`="noreply@humhub.com"

Reply:
Also moved env.php file functionality to separate helper for unit test. Tests not ready yet

Reply:
@gevorgmansuryan Thank you, I like that.

But I hoped that we could now simply overwrite all config values via the ENV variables. e.g.
`HUMHUB_CONFIG_COMPONENTS_URLMANAGER_SHOWSCRIPTNAME=false`

But if I understand correctly, it's not possible because of Upper/LowerCase, is it?

If we use this syntax it should work right? 
`HUMHUB_CONFIG_components_urlManager_showScriptName=false`
`HUMHUB_CONFIG_params_fixedSettings_base_mailer.transportType=php`

Reply:
@luke- I can enable this too. currently there is 2 namespaces
HUMHUB_PARAMS - it's config->params
HUMHUB_SETTINGS - fixed settings

i can add HUMHUB_COMPONENTS namespace too and this will work also
HUMHUB_COMPONENTS.URL_MANAGER.SHOW_SCRIPT_NAME=false

Reply:
or we can have one namespace for all

`HUMHUB_CONFIG.PARAMS.MODULE_AUTOLOAD_PATHS`=["/app/modules/humhub","/app/modules/humhub-contrib"]
`HUMHUB_CONFIG.PARAMS.BASE.MAILER.SYSTEM_EMAIL_ADDRESS`="[noreply@humhub.com](mailto:noreply@humhub.com)"
`HUMHUB_CONFIG.COMPONENTS.URL_MANAGER.SHOW_SCRIPT_NAME`=false

Reply:
@gevorgmansuryan One namespace is probably best.
But how does the conversation from UpperCase to camelCase work?

I'm not quite sure how you want to automatically convert `HUMHUB_CONFIG.COMPONENTS.URL_MANAGER.SHOW_SCRIPT_NAME` to `$config[‘components’][‘urlManager’][‘showScriptName’]`.

Reply:
@luke- I'm using yii\helpers\Inflector::variablize to do this. but there is one problem, both env and variablize do not support `-` symbol. So with one namespace we can not handle 'fixed-settings' or any other config key with `-` symbol. Let's keep 2 namespaces HUMHUB_SETTINGS for fixed settings, and `HUMHUB_CONFIG` for everything

Reply:
@gevorgmansuryan Ok, that looks like a good solution!

So we use:
- `.` for the configuration depth / array
- `_` to convert uppercase vars into camelcase

So we could cover the manipulation of the most config array values. Do you see config array elements that cannot be manipulated via ENV or other restrictions? That would be good for the documentation.

---



Please use `HUMHUB_FIXED_SETTINGS` as a second namespace. These FS are rarely used and should be clearly labelled.

Reply:
@luke- It covers almost evrything, Yii 100% keeps the `lowerCamelCase` in base configs. 
.env does not support space and dots. For example `['components']['db']['on afterOpen']`. But it can be done using json, like this:
HUMHUB_CONFIG.COMPONENTS.DB = {"on afterOpen": ["humhub\libs\Helpers", "SqlMode"]}

Reply:
Finally. 
this:
![image](https://github.com/user-attachments/assets/ad76109c-3395-4f66-b8f9-4285a61860ff)
will be converted to this:
![image](https://github.com/user-attachments/assets/5762d7fb-fc30-46e1-9af8-4bee582822a6)